### DISCUSSION TITLE
Implementation of uncertain LMI

### DISCUSSION THREAD

**Discussion by MT8444** [yalmip/YALMIP#1448] [General discussion] [answered]
I am trying to implement stability analysis for LPV model, inspired from this example : https://yalmip.github.io/example/lpvstatefeedback/

I have two related questions. Consider this simple example : 

```matlab

nx = 2;
delta = sdpvar(1);
A = [0 1;-2 -3+delta];

P = sdpvar(nx);
cs = (P>=1e-3);
cs = [cs, (A'*P+P*A) <= -1e-3];

s = 2;
cs = [cs, -s <= delta <= s, uncertain(delta)];

opt = sdpsettings('solver','mosek');
sol = optimize(cs,[],opt);
``` 
This shows feasibility as long as s < 3, as expected. 

I was however wondering how is implemented the uncertain set declaration. If I replace above condition with an inconsistent one : 

```matlab

cs = [cs, 1 <= delta <= -2, uncertain(delta)];
```
this show feasibility without any warning. What exactly do represent the "coefficient range" of the associated element-wise inequality ? It shows "1 to 2" for the 2 examples listed above.

Another, yet less important, question concerns the underlying implementation of the robust LMI. Is the above example exactly equivalent to following the 2-vertices LMI problem ? Is it what is done when the robustification module displays "Enumerated 2 vertices" ? 

```matlab

% consider -1 <= s <= 2
nx = 2;
A1 = [0 1;-2 -4];
A2 = [0 1;-2 -1];

P = sdpvar(nx);
cs = (P>=1e-3);
cs = [cs, (A1'*P+P*A1) <= -1e-3];
cs = [cs, (A2'*P+P*A2) <= -1e-3];

opt = sdpsettings('solver','mosek');
sol = optimize(cs,[],opt);
``` 

Thanks in advance for your help !

**Comment by johanlofberg** [comment, marked as answer]
The code simply doesn't account for the completely inconsistent case so the result is unknown. It assumes a sensible uncertainty model

BTW, you write 1e-3 but most likely mean 1e-3*eye(2)

**Reply by MT8444** [reply]
Ok, thank you. But in this case what does the "coefficient range" stand for ? (for the valid case)

**Comment by johanlofberg** [comment]
Coefficient range is used to understand the numerical conditioning of an expression

```
>> sdpvar x y z
x - 2e-5*y + 10^10*z
Linear scalar (real, 3 variables)
Coefficients range: 2e-05 to 10000000000
```
